httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/DefaultBHttpClientConnectionFactory.java

```diff
@@ -100,10 +120,87 @@ public DefaultBHttpClientConnection createConnection(final Socket socket) throws
                 CharCodingSupport.createEncoder(this.charCodingConfig),
                 this.incomingContentStrategy,
                 this.outgoingContentStrategy,
+                this.responseOutOfOrderStrategy,
                 this.requestWriterFactory,
                 this.responseParserFactory);
         conn.bind(socket);
         return conn;
     }
 
+    @Override
+    public String toString() {
```

Comment:
Strictly speaking, this should be a separate PR.

Reply:
I can remove this if you like, would you prefer a separate ticket for `DefaultBHttpClientConnectionFactory.toString()`?

Reply:
Sorry for the confusion. The `toString` isn't the problem. Although, I see little benefit over the autogenerated one, but the builder introduction. This should be a separate PR.

Reply:
If I remove the builder from this PR I'll need to either sequence the builder PR prior to this one or introduce another public constructor. What do you prefer?

Reply:
Let's do the builder first and the modify it in this PR.